src/handlers/delegatedStakingHandler.ts

```diff
@@ -0,0 +1,192 @@
+  rewards: change.rewards,
+  timestamp: change.created_at,
+});
+
+const delegateStakeBalanceChangeResponses = (txs) =>
+  txs.map(delegateStakeBalanceChangeResponse);
+
+export const delegatedStakes = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        delegate_stake_withdraw_event: null,
+      },
+      orderBy: [{ ordinal: "desc" }],
+    },
+    prisma.delegate_stake_create_events.findMany,
+    delegateStakeCreateResponses
+  );
+};
+
+export const delegatedStake = async (event) => {
+  try {
+    const hash = event.pathParameters?.hash;
+
+    const stake = await prisma.delegate_stake_create_events.findUnique({
+      where: { hash },
+    });
+
+    return respond(stake, delegateStakeCreateResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const addressDelegatedStakes = async (event) => {
+  const address = event.pathParameters?.address;
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        source_addr: address,
+      },
+      orderBy: [{ ordinal: "desc" }],
+    },
+    prisma.delegate_stake_create_events.findMany,
+    delegateStakeCreateResponses
+  );
+};
+
+export const delegatedStakeWithdrawals = async (event) => {
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      orderBy: [{ created_at: "desc" }],
+    },
+    prisma.delegate_stake_withdraw_events.findMany,
+    delegateStakeWithdrawResponses
+  );
+};
+
+export const delegatedStakeWithdrawal = async (event) => {
+  try {
+    const hash = event.pathParameters?.hash;
+
+    const withdrawal = await prisma.delegate_stake_withdraw_events.findUnique({
+      where: { hash },
+    });
+
+    return respond(withdrawal, delegateStakeWithdrawResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const addressDelegatedStakeWithdrawals = async (event) => {
+  const address = event.pathParameters?.address;
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        source_addr: address,
+      },
+      orderBy: [{ created_at: "desc" }],
+    },
+    prisma.delegate_stake_withdraw_events.findMany,
+    delegateStakeWithdrawResponses
+  );
+};
+
+export const stakingBalanceByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const { ordinal } = event.queryStringParameters || {};
+
+    const ordinalNbr = toNumber(ordinal);
+
+    const ordinalCondition = isFinite(ordinalNbr)
+      ? { global_snapshot_ordinal: { lte: ordinalNbr } }
+      : {};
+
+    const latestPerNode = await prisma.delegate_stake_balance_changes.groupBy({
```

Comment:
When a position is withdrawn, these records will still exist for the NodeID but have a zero balance right? I think we should filter those out by default.